internal/builders/generic/README.md

```diff
@@ -631,7 +632,82 @@ jobs:
 
   # =========================================================
   #
-  # Step 5: Call the generic workflow to generate provenance
+  # Step 4: Call the generic workflow to generate provenance
+  #         by declaring the job below.
+  #
+  # =========================================================
+  provenance:
+    needs: [build]
+    permissions:
+      actions: read   # To read the workflow path.
+      id-token: write # To sign the provenance.
+      contents: write # To add assets to a release.
+    uses: slsa-framework/slsa-github-generator/.github/workflows/generator_generic_slsa3.yml@v1.2.0
+    with:
+      base64-subjects: "${{ needs.build.outputs.hashes }}"
+      upload-assets: true # Optional: Upload to a new release
+```
+
+### Provenance for Haskell
+
+If you use [Haskell](https://www.haskell.org/) (either via
+[`cabal`](https://www.haskell.org/cabal/) or
+[`stack`](https://docs.haskellstack.org/en/stable/README/)) to generate your
+artifacts, you can easily generate SLSA3 provenance by updating your existing
+workflow with the 5 steps indicated in the workflow below.
+
+```yaml
+jobs:
+  build:
+    # ==================================================
+    #
+    # Step 1: Declare an `outputs` for the hashes to be
+    #         used during the provenance steps.
+    #
+    # ==================================================
+    outputs:
+      hashes: ${{ steps.hash.outputs.hashes }}
+
+    [...]
+
+    steps:
+      [...]
+      - name: Build using Haskell
+        # =================================================
+        #
+        # Step 2: Add an `id: build` field
+        #         to your build step
+        #
+        # =================================================
+        id: build
```

Comment:
you can remove the `id: build` and the Step 2 comment above; and re-number the next comments.
The id was needed for goreleaser but is not for Haskel. (I'm going to send a PR removing it for builders that don't need it as I just realized it)

Reply:
Was just wondering where it was used today. I'll send a commit for this one.